### PR TITLE
Working test framework and initial set of tests

### PR DESCRIPTION
**Pull request by Defimatt**
Hey @binzydev, I don't know at which point you'd want to merge this, but this is a working test framework with some initial tests for TirRune and TirChef. 

I'll add tests to get more coverage as we go along, but I think this is a good proof of concept.

Have updated package.json npm scripts to use hardhat for tests and coverage, but have left the rest alone so the build process should remain the same.

If you want me to get a bit further with test coverage before merging, let me know!

### CODE REVIEW

File: test/TirChef.test.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+    before(async() => {   
+        const signers = await ethers.getSigners();
+        alice = signers[0];
+        bob = signers[1];
+        carol = signers[2];
+        dev = signers[3];
+        vault = signers[4];
+        charity = signers[5];
+        bot = signers[6];
```
Comment (binzydev) [inline test/TirChef.test.ts:57]: const { alice, bob, carol, dev, vault, charity, bot } = await ethers.getSigners();

File: test/TirChef.test.ts [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    });
+
+    it("should default to zero fees", async() => {
+        expect(await TirChef["devMintPercent()"]()).to.equal(0);
+        expect(await TirChef["vaultMintPercent()"]()).to.equal(0);
+        expect(await TirChef["charityMintPercent()"]()).to.equal(0);
+        expect(await TirChef["devDepositPercent()"]()).to.equal(0);
+        expect(await TirChef["vaultDepositPercent()"]()).to.equal(0);
+        expect(await TirChef["charityDepositPercent()"]()).to.equal(0);
+    });
```
Comment (binzydev) [inline test/TirChef.test.ts:113]: It's a nice test but I would typically only test that they do what they say they should, otherwise it's more work to maintain and they could mislead you to believe they actually work

File: test/TirChef.test.ts [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+            await TirChef.add(100, LP2.address, 100, true);
+
+            await LP2AsAlice.approve(TirChef.address, 1000);
+            await TirChefAsAlice.deposit(1, 100);
+            expect(await LP2.balanceOf(alice.address)).to.equal(900);
+            
+            await TirChefAsAlice.emergencyWithdraw(1);
+            expect(await LP2.balanceOf(alice.address)).to.equal(999);
+            expect(await TirRune.balanceOf(alice.address)).to.equal(0); // Check no Tir rewards
+        });
```
Comment (binzydev) [inline test/TirChef.test.ts:164]: Love it

File: test/TirChef.test.ts (hunk 2 above)
Comment (MattDuffin) [reply test/TirChef.test.ts:113]: Absolutely, I was planning on adding additional tests once they've been set to a non-zero value, calling deposit and minting rewards, then checking they get the right ratio of fees. This was just to make sure they have the correct initial values.

**Comment by MattDuffin** [comment]
@binzydev I got an email with this in:

`const { alice, bob, carol, dev, vault, charity, bot } = await ethers.getSigners();`

But didn't find it in the review comments.

Good shout, I forgot about object destructuring, makes life easier.

**Comment by binzydev** [comment]
Looks like these are in 👍 ⚡

**Comment by Defimatt** [comment]
> Looks like these are in 👍 ⚡

Nice, my first contrib! Hopefully first of many 😄
